In [1]:
import pandas as pd
df = pd.read_csv("bridges.csv")

In [2]:
df. head()

,Id,river,location,erected,purpose,length,lanes,clear-g,t-or-d,material,span,rel-l,type
0,E1,M,3.0,1818,HIGHWAY,NaN,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
1,E2,A,25.0,1819,HIGHWAY,1037.0,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
2,E3,A,39.0,1829,AQUEDUCT,NaN,1.0,N,THROUGH,WOOD,NaN,S,WOOD
3,E5,A,29.0,1837,HIGHWAY,1000.0,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
4,E6,M,23.0,1838,HIGHWAY,NaN,2.0,N,THROUGH,WOOD,NaN,S,WOOD


In [3]:
df. shape

(108, 13)

In [4]:
df. info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 108 entries, 0 to 107
Data columns (total 13 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Id        108 non-null    object 
 1   river     108 non-null    object 
 2   location  107 non-null    float64
 3   erected   108 non-null    int64  
 4   purpose   108 non-null    object 
 5   length    81 non-null     float64
 6   lanes     92 non-null     float64
 7   clear-g   106 non-null    object 
 8   t-or-d    102 non-null    object 
 9   material  106 non-null    object 
 10  span      92 non-null     object 
 11  rel-l     103 non-null    object 
 12  type      106 non-null    object 
dtypes: float64(3), int64(1), object(9)
memory usage: 11.1+ KB


In [5]:
df.describe()

,location,erected,length,lanes
count,107.000000,108.000000,81.000000,92.000000
mean,25.978505,1905.314815,1567.469136,2.630435
std,13.665871,37.173664,747.491523,1.164657
min,1.000000,1818.000000,804.000000,1.000000
25%,15.500000,1884.000000,1000.000000,2.000000
50%,27.000000,1903.000000,1300.000000,2.000000
75%,37.500000,1928.000000,2000.000000,4.000000
max,52.000000,1986.000000,4558.000000,6.000000


**Plausibility Assumption:** I will consider bridge lengths greater than 2500 to potentially extreme and investigate those  values further. I chose 2500 because 75% of the bridge lengths are 2000  or lower, so values above 2500 are unsually high compared with the rest of the dataset.

In [6]:
df[df["length"] >2500]

,Id,river,location,erected,purpose,length,lanes,clear-g,t-or-d,material,span,rel-l,type
31,E34,O,41.0,1888,RR,4558.0,2.0,G,THROUGH,STEEL,LONG,F,SIMPLE-T
44,E46,A,37.0,1897,RR,4000.0,2.0,G,DECK,STEEL,LONG,F,SIMPLE-T
52,E87,A,35.0,1903,RR,3000.0,2.0,G,THROUGH,STEEL,MEDIUM,S,SIMPLE-T
65,E61,O,41.0,1915,RR,2822.0,2.0,G,THROUGH,STEEL,LONG,F,SIMPLE-T
76,E72,M,5.0,1927,HIGHWAY,2663.0,4.0,N,DECK,STEEL,MEDIUM,S-F,CANTILEV
78,E75,A,30.0,1928,HIGHWAY,2678.0,4.0,G,DECK,STEEL,MEDIUM,F,ARCH
104,E91,O,44.0,1975,HIGHWAY,3756.0,6.0,G,THROUGH,STEEL,LONG,F,ARCH


**Flagged values:** Using my cutoff of 2500 for length, seven values were flagged for investigation: 4558, 4000, 3000, 2822, 2663, 2678, and 3756. After reviewing the flagged values, I consider 3000, 3756,4000 and 4558 to be extreme values but still possible. I decided to keep these values unchanged because I don't have any evidence to determine that they are errors. 

Git Hub Link: https://github.com/jazzminowens/cs82a-portfolio